testing scripts fails with python 3.5

The failing test is:
pyFAI.test.test_bug_regression.TestBug211
```
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.5/pyFAI/test/test_bug_regression.py", line 149, in test_quantile
    self.fail()
AssertionError: None
```
which means this command failed: 
```/usr/bin/python3 /usr/bin/pyFAI-average --quiet -q 0.2-0.8 ...```
with:
```
Traceback (most recent call last):
  File "/usr/bin/pyFAI-average", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3019, in <module>
    @_call_aside
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3003, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3032, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 655, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 963, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 849, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pyFAI==0.15.0a1' distribution was not found and is required by the application
```
On the other hand:
```/usr/bin/pyFAI-average --quiet -q 0.2-0.8 ...```
works.

Is this test still valid ? because the tested script is not the right one.

--- a/pyFAI/test/test_bug_regression.py
+++ b/pyFAI/test/test_bug_regression.py
@@ -117,11 +117,10 @@ class TestBug211(unittest.TestCase):
             e.write(fn)
             self.image_files.append(fn)
         self.res = res / 3.0
-        self.exe, self.env = UtilsTest.script_path("pyFAI-average")
         # It is not anymore a script, but a module
-        if not os.path.exists(self.exe):
-            import pyFAI.app.average
-            self.exe = pyFAI.app.average.__file__
+        import pyFAI.app.average
+        self.exe = pyFAI.app.average.__file__
+        self.env = UtilsTest.get_test_env()
 
     def tearDown(self):
         for fn in self.image_files:

--- a/pyFAI/test/test_scripts.py
+++ b/pyFAI/test/test_scripts.py
@@ -34,12 +34,13 @@ __author__ = "Valentin Valls"
 __contact__ = "valentin.valls@esrf.fr"
 __license__ = "MIT"
 __copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
-__date__ = "10/01/2018"
+__date__ = "24/01/2018"
 
 import sys
 import unittest
-import runpy
 import logging
+import subprocess
+from pyFAI.test.utilstest import UtilsTest
 
 logger = logging.getLogger(__name__)
 
@@ -56,67 +57,89 @@ except ImportError:
 
 class TestScriptsHelp(unittest.TestCase):
 
-    def executeAppHelp(self, module):
-        old_sys_argv = list(sys.argv)
-        try:
-            sys.argv = [None, "--help"]
-            runpy.run_module(mod_name=module, run_name="__main__", alter_sys=True)
-        except SystemExit as e:
-            self.assertEquals(e.args[0], 0)
-        finally:
-            sys.argv = old_sys_argv
+    def executeCommandLine(self, command_line, env):
+        """Execute a command line.
+
+        Log output as debug in case of bad return code.
+        """
+        logger.info("Execute: %s", " ".join(command_line))
+        p = subprocess.Popen(command_line,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             env=env)
+        out, err = p.communicate()
+        logging.info("Return code: %d", p.returncode)
+        if p.returncode != 0:
+            try:
+                out = out.decode('utf-8')
+            except UnicodeError:
+                pass
+            try:
+                err = err.decode('utf-8')
+            except UnicodeError:
+                pass
+            logging.debug("stdout:")
+            logging.debug("%s", out)
+            logging.debug("stderr:")
+            logging.debug("%s", err)
+
+    def executeAppHelp(self, script_name, module_name):
+        script = UtilsTest.script_path(script_name, module_name)
+        env = UtilsTest.get_test_env()
+        command_line = [sys.executable, script, "--help"]
+        self.executeCommandLine(command_line, env)
 
     def testCheckCalib(self):
         if qt is None:
             self.skipTest("Library Qt is not available")
-        self.executeAppHelp("pyFAI.app.check_calib")
+        self.executeAppHelp("check_calib", "pyFAI.app.check_calib")
 
     def testDetector2Nexus(self):
-        self.executeAppHelp("pyFAI.app.detector2nexus")
+        self.executeAppHelp("detector2nexus", "pyFAI.app.detector2nexus")
 
     def testDiffMap(self):
-        self.executeAppHelp("pyFAI.app.diff_map")
+        self.executeAppHelp("diff_map", "pyFAI.app.diff_map")
 
     def testDiffTomo(self):
-        self.executeAppHelp("pyFAI.app.diff_tomo")
+        self.executeAppHelp("diff_tomo", "pyFAI.app.diff_tomo")
 
     def testEigerMask(self):
-        self.executeAppHelp("pyFAI.app.eiger_mask")
+        self.executeAppHelp("eiger-mask", "pyFAI.app.eiger_mask")
 
     def testMxcalibrate(self):
         if qt is None:
             self.skipTest("Library Qt is not available")
-        self.executeAppHelp("pyFAI.app.mx_calibrate")
+        self.executeAppHelp("MX-calibrate", "pyFAI.app.mx_calibrate")
 
     def testPyfaiAverage(self):
-        self.executeAppHelp("pyFAI.app.average")
+        self.executeAppHelp("pyFAI-average", "pyFAI.app.average")
 
     def testPyfaiBenchmark(self):
-        self.executeAppHelp("pyFAI.app.benchmark")
+        self.executeAppHelp("pyFAI-benchmark", "pyFAI.app.benchmark")
 
     def testPyfaiCalib(self):
         if qt is None:
             self.skipTest("Library Qt is not available")
-        self.executeAppHelp("pyFAI.app.calib")
+        self.executeAppHelp("pyFAI-calib", "pyFAI.app.calib")
 
     def testPyfaiDrawmask(self):
         if qt is None or silx is None:
             self.skipTest("Library Qt and silx are not available")
-        self.executeAppHelp("pyFAI.app.drawmask")
+        self.executeAppHelp("pyFAI-drawmask", "pyFAI.app.drawmask")
 
     def testPyfaiIntegrate(self):
-        self.executeAppHelp("pyFAI.app.integrate")
+        self.executeAppHelp("pyFAI-integrate", "pyFAI.app.integrate")
 
     def testPyfaiRecalib(self):
         if qt is None:
             self.skipTest("Library Qt is not available")
-        self.executeAppHelp("pyFAI.app.recalib")
+        self.executeAppHelp("pyFAI-recalib", "pyFAI.app.recalib")
 
     def testPyfaiSaxs(self):
-        self.executeAppHelp("pyFAI.app.saxs")
+        self.executeAppHelp("pyFAI-saxs", "pyFAI.app.saxs")
 
     def testPyfaiWaxs(self):
-        self.executeAppHelp("pyFAI.app.waxs")
+        self.executeAppHelp("pyFAI-waxs", "pyFAI.app.waxs")
 
 
 def suite():

--- a/pyFAI/test/test_scripts.py
+++ b/pyFAI/test/test_scripts.py
@@ -69,19 +69,26 @@ class TestScriptsHelp(unittest.TestCase):
                              env=env)
         out, err = p.communicate()
         logging.info("Return code: %d", p.returncode)
+        try:
+            out = out.decode('utf-8')
+        except UnicodeError:
+            pass
+        try:
+            err = err.decode('utf-8')
+        except UnicodeError:
+            pass
+
         if p.returncode != 0:
-            try:
-                out = out.decode('utf-8')
-            except UnicodeError:
-                pass
-            try:
-                err = err.decode('utf-8')
-            except UnicodeError:
-                pass
+            logging.info("stdout:")
+            logging.info("%s", out)
+            logging.info("stderr:")
+            logging.info("%s", err)
+        else:
             logging.debug("stdout:")
             logging.debug("%s", out)
             logging.debug("stderr:")
             logging.debug("%s", err)
+        self.assertEqual(p.returncode, 0)
 
     def executeAppHelp(self, script_name, module_name):
         script = UtilsTest.script_path(script_name, module_name)

--- a/pyFAI/test/utilstest.py
+++ b/pyFAI/test/utilstest.py
@@ -29,7 +29,7 @@ __author__ = "Jérôme Kieffer"
 __contact__ = "jerome.kieffer@esrf.eu"
 __license__ = "MIT"
 __copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
-__date__ = "12/01/2018"
+__date__ = "24/01/2018"
 
 PACKAGE = "pyFAI"
 DATA_KEY = "PYFAI_DATA"
@@ -124,6 +124,14 @@ class TestContext(object):
             parser.add_argument(dest="args", type=str, nargs='*')
             self.options = parser.parse_args([])
         return self.options
+
+    def get_test_env(self):
+        """
+        Returns an associated environment with a working project.
+        """
+        env = dict((str(k), str(v)) for k, v in os.environ.items())
+        env["PYTHONPATH"] = os.pathsep.join(sys.path)
+        return env
 
     def script_path(self, script):
         """

--- a/pyFAI/test/utilstest.py
+++ b/pyFAI/test/utilstest.py
@@ -81,6 +81,7 @@ class TestContext(object):
         self.reloaded = False
         self.name = PACKAGE
         self.script_dir = None
+        self.installed = False
 
         self.download_images = self.resources.download_all
         self.getimage = self.resources.getfile
@@ -133,7 +134,17 @@ class TestContext(object):
         env["PYTHONPATH"] = os.pathsep.join(sys.path)
         return env
 
-    def script_path(self, script):
+    def script_path(self, script_name, module_name):
+        """Returns the script path according to it's location"""
+        if self.installed:
+            script = self.get_installed_script_path(script_name)
+        else:
+            import importlib
+            module = importlib.import_module(module_name)
+            script = module.__file__
+        return script
+
+    def get_installed_script_path(self, script):
         """
         Returns the path of the executable and the associated environment
 
@@ -145,12 +156,7 @@ class TestContext(object):
         else:
             available_extensions = [""]
 
-        env = dict((str(k), str(v)) for k, v in os.environ.items())
-        env["PYTHONPATH"] = os.pathsep.join(sys.path)
         paths = os.environ.get("PATH", "").split(os.pathsep)
-        if self.script_dir is not None:
-            paths.insert(0, self.script_dir)
-
         for base in paths:
             # clean up extra quotes from paths
             if base.startswith('"') and base.endswith('"'):
@@ -160,11 +166,11 @@ class TestContext(object):
                 print(script_path)
                 if os.path.exists(script_path):
                     # script found
-                    return script_path, env
+                    return script_path
         # script not found
         logger.warning("Script '%s' not found in paths: %s", script, ":".join(paths))
         script_path = script
-        return script_path, env
+        return script_path
 
     def _initialize_tmpdir(self):
         """Initialize the temporary directory"""

--- a/run_tests.py
+++ b/run_tests.py
@@ -32,7 +32,7 @@ Test coverage dependencies: coverage, lxml.
 """
 
 __authors__ = ["Jérôme Kieffer", "Thomas Vincent"]
-__date__ = "10/01/2018"
+__date__ = "24/01/2018"
 __license__ = "MIT"
 
 import distutils.util
@@ -401,6 +401,7 @@ if old_importer:
     print(dir(test_module))
     utilstest = getattr(test_module, "utilstest")
 UtilsTest = utilstest.UtilsTest
+UtilsTest.installed = options.installed
 
 if options.low_mem:
     logger.info("Switch to low_mem mode")

